src/THORChain/Swap.cpp

```diff
@@ -71,7 +71,31 @@ TWCoinType chainCoinType(Chain chain) {
     }
 }
 
-std::string chainName(Chain chain) {
+TWCoinType chainCoinTypeMAYA(Chain chain) {
+    switch (chain) {
+    case Chain::ETH:
+        return TWCoinTypeEthereum;
+    case Chain::BTC:
+        return TWCoinTypeBitcoin;
+    case Chain::THOR:
+        return TWCoinTypeTHORChain;
+    case Chain::DASH:
+        return TWCoinTypeDash;
+    case Chain::MAYA:
+    default:
+        return TWCoinTypeMAYAChain;
+    }
+}
+
+TWCoinType chainCoinType(Chain toChain, Chain mfromChain) {
+    if(mfromChain == Chain::MAYA){
+        return chainCoinTypeMAYA(toChain);
```

Comment:
Do I understand correctly that MAYASwap is only needed to swap `CACAO->X`, and not vice versa?

Reply:
`MAYASwap` should be able to swap:

1. CACAO -> ASSET
2. ASSET -> CACAO
3. ASSET A -> ASSET B